client/asset/eth/rpcclient.go

```diff
@@ -0,0 +1,215 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package eth
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+// Check that rpcclient satisfies the ethFetcher interface.
+var _ ethFetcher = (*rpcclient)(nil)
+
+// rpcclient satisfies the ethFetcher interface. Do not use until Connect is
+// called.
+type rpcclient struct {
+	// c is a direct client for raw calls.
+	c *rpc.Client
+	// ec wraps the client with some useful calls.
+	ec *ethclient.Client
+	n  *node.Node
+}
+
+// connect connects to a node. It then wraps ethclient's client and
+// bundles commands in a form we can easily use.
+func (c *rpcclient) connect(ctx context.Context, node *node.Node, contractAddr common.Address) error { // contractAddr will be used soonTM
```

Comment:
It's been a while, and my memory is a little fuzzy, but it may be possible to make it an argument. I think the client needs access to the actual contract's code from startup, so any attempt to use a contract that isn't already hard coded should fail. So, it is effectively a constant, and might as well be parsed once at startup. The address could also be a constant, really, it always ends up being the same on simnet anyway, being sent by the same account with the same nonce, I believe. This is used in #1019 so sooner than soon.